checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -432,10 +433,29 @@ private UBQualifier removeArrayLengths(LessThanLengthOf i, LessThanLengthOf j) {
     public TransferResult<CFValue, CFStore> visitFieldAccess(
             FieldAccessNode n, TransferInput<CFValue, CFStore> in) {
         if (isArrayLengthFieldAccess(n)) {
+
             FieldAccess arrayLength = FlowExpressions.internalReprOfFieldAccess(atypeFactory, n);
             Receiver arrayRec = arrayLength.getReceiver();
+
+            // Look up the SameLen type of the array.
+            Tree arrayTree = n.getReceiver().getTree();
+            AnnotationMirror sameLenAnno = atypeFactory.sameLenAnnotationFromTree(arrayTree);
+            List<String> sameLenArrays =
+                    sameLenAnno == null
+                            ? new ArrayList<String>()
+                            : IndexUtil.getValueOfAnnotationWithStringArgument(sameLenAnno);
+
+            if (!sameLenArrays.contains(arrayRec.toString())) {
+                sameLenArrays.add(arrayRec.toString());
+            }
+
+            ArrayList<String> negativeOnes = new ArrayList<>(sameLenArrays.size());
```

Comment:
I would call this `offsets` which reflects its semantics rather than `negativeOnes` which reflects its contents.